crates/primitives/src/env.rs

```diff
@@ -46,37 +70,124 @@ pub struct OptimismFields {
     pub enveloped_tx: Option<Bytes>,
 }
 
+impl BlockEnv {
+    /// See [EIP-4844] and [`Env::calc_data_fee`].
+    ///
+    /// Returns `None` if `Cancun` is not enabled. This is enforced in [`Env::validate_block_env`].
+    ///
+    /// [EIP-4844]: https://eips.ethereum.org/EIPS/eip-4844
+    #[inline]
+    pub fn get_blob_gasprice(&self) -> Option<u64> {
+        self.excess_blob_gas.map(calc_blob_fee)
+    }
+}
+
+/// The transaction environment.
 #[derive(Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct TxEnv {
-    /// Caller or Author or tx signer
+    /// The caller, author or signer of the transaction.
     pub caller: B160,
+    /// The gas limit of the transaction.
     pub gas_limit: u64,
+    /// The gas price of the transaction.
     pub gas_price: U256,
-    pub gas_priority_fee: Option<U256>,
+    /// The destination of the transaction.
     pub transact_to: TransactTo,
+    /// The value sent to `transact_to`.
     pub value: U256,
+    /// The data of the transaction.
     #[cfg_attr(feature = "serde", serde(with = "crate::utilities::serde_hex_bytes"))]
     pub data: Bytes,
-    pub chain_id: Option<u64>,
+    /// The nonce of the transaction. If set to `None`, no checks are performed.
     pub nonce: Option<u64>,
+
+    /// The chain ID of the transaction. If set to `None`, no checks are performed.
+    ///
+    /// Incorporated as part of the Spurious Dragon upgrade via [EIP-155].
+    ///
+    /// [EIP-155]: https://eips.ethereum.org/EIPS/eip-155
+    pub chain_id: Option<u64>,
+
+    /// A list of addresses and storage keys that the transaction plans to access.
+    ///
+    /// Added in [EIP-2930].
+    ///
+    /// [EIP-2930]: https://eips.ethereum.org/EIPS/eip-2930
     pub access_list: Vec<(B160, Vec<U256>)>,
+
```

Comment:
Conflicts in the env for 4844 support; Not too hard to resolve, new fields for blob txs.